src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -302,18 +306,31 @@ private void doWriteDates(Type type, ClickHousePipedOutputStream stream, Data va
                 if (value.getFieldType().equals(Schema.Type.INT64)) {
                     if (value.getObject().getClass().getName().endsWith(".Date")) {
                         Date date = (Date) value.getObject();
-                        long time = date.getTime();
-                        BinaryStreamUtils.writeInt64(stream, time);
+                        BinaryStreamUtils.writeInt64(stream, date.getTime());
                     } else {
                         BinaryStreamUtils.writeInt64(stream, (Long) value.getObject());
                     }
                 } else if (value.getFieldType().equals(Schema.Type.STRING)) {
                     try {
-                        ZonedDateTime zonedDateTime = ZonedDateTime.parse((String) value.getObject());
-                        long seconds = zonedDateTime.toInstant().getEpochSecond();
-                        long milliSeconds = zonedDateTime.toInstant().toEpochMilli();
-                        long microSeconds = TimeUnit.MICROSECONDS.convert(seconds, TimeUnit.SECONDS) + zonedDateTime.get(ChronoField.MICRO_OF_SECOND);
-                        long nanoSeconds = TimeUnit.NANOSECONDS.convert(seconds, TimeUnit.SECONDS) + zonedDateTime.getNano();
+                        long seconds;
+                        long milliSeconds;
+                        long microSeconds;
+                        long nanoSeconds;
+
+                        if (!csc.getDateTimeFormat().isEmpty()) {
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(csc.getDateTimeFormat());
```

Comment:
Maybe we can initialize the `formatter` only once -> also we need to explain that we only support one format (how we plan to deal if we have different formats for different variables )